File: pys/yakup.py
import re
import requests
from bs4 import BeautifulSoup

url = 'https://www.yakup.com/news/index.html?cat=all'

req = requests.get(url)
bs = BeautifulSoup(req.text, 'lxml')

wrapper = bs.find('ul', class_='tp_5')
items = wrapper.find_all('div', class_='listBoxType_3')

for item in items[:1]:
    a_tag = item.find('a')
    title = a_tag.get_text().strip()
    href = 'https://www.yakup.com' + a_tag['href']
    span = item.find('p', class_='number tp_3').get_text()
    date = re.search('[0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2}', span).group()

    req2 = requests.get(href)
    bs2 = BeautifulSoup(req2.content, 'lxml')
    wrapper2 = bs2.find('div', class_='bodyarea')

    contents = ''.join([it.get_text().strip() for it in wrapper2.find_all('p')]).replace('\n', '')

    print(title, href, date, contents)

File: crawlers/yakup.go
package crawlers

import (
	"fmt"
	"net/http"

	"flashnews/models"
	"flashnews/utils"

	"github.com/PuerkitoBio/goquery"
)

const yakupCommonURL = "http://m.yakup.com"
const yakupItemURL = "http://m.yakup.com"

type Yakup struct{}

func (c Yakup) GetName() string {
	return "yakup"
}

func (c Yakup) GetList(number int) ([]models.NewsItem, error) {
	// Number
	var _number int
	if number > 15 || number < 1 {
		_number = 15
	} else {
		_number = number
	}
	result := make([]models.NewsItem, _number)

	// Request
	req, err := http.Get(yakupCommonURL)
	if err != nil {
		return result, err
	}
	defer req.Body.Close()

	if req.StatusCode != 200 {
		return result, fmt.Errorf("[ERROR] Request Status Code : %d, %s", req.StatusCode, req.Status)
	}

	// HTML Load
	html, err := goquery.NewDocumentFromReader(req.Body)
	if err != nil {
		return result, err
	}

	// Parsing
	cnt := 0
	wrappers := html.Find("div.box-list-01")
	wrappers.Each(func(i int, sel *goquery.Selection) {
		if i >= 3 {
			return
		}

		items := sel.Find("li")
		items.Each(func(j int, sel2 *goquery.Selection) {
			if cnt >= _number {
				return
			}
			aTag := sel2.Find("a")
			href, ok := aTag.Attr("href")
			if !ok {
				result[cnt] = models.NewsItem{}
				return
			}
			url := yakupItemURL + href[1:]

			title := utils.TrimAll(aTag.Text())

			result[cnt] = models.NewsItem{
				Title:    title,
				URL:      url,
				Contents: "",
				Datetime: "",
			}
			cnt++
		})
	})

	return result, nil
}

func (c Yakup) GetContents(item *models.NewsItem) error {
	// Request
	req, err := http.Get(item.URL)
	if err != nil {
		return err
	}
	defer req.Body.Close()

	if req.StatusCode != 200 {
		return fmt.Errorf("[ERROR] Request Status Code : %d, %s", req.StatusCode, req.Status)
	}

	// HTML Load
	html, err := goquery.NewDocumentFromReader(req.Body)
	if err != nil {
		return err
	}

	// Date
	dtTags := html.Find("dl.subdata").Find("dt")
	dtTags.Each(func(i int, sel *goquery.Selection) {
		if i == 2 {
			item.Datetime = sel.Text()
		}
	})

	// Parsing
	wrapper := html.Find("div.article_view")
	contents := utils.TrimAll(wrapper.Text())

	item.Contents = contents
	return nil
}
